step-controller/step-controller-server/src/main/java/step/core/execution/ExecutionServices.java

```diff
@@ -263,21 +266,43 @@ public AggregatedReportView getAggregatedReportView(@PathParam("id") String exec
 	@GET
 	@Path("/{id}/report/{customReportType}")
 	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.TEXT_PLAIN)
+	@Secured(right = "execution-read")
+	public Response getCustomReport(@PathParam("id") String executionId, @PathParam("customReportType") String customReportType) {
+		ExecutionEngineContext executionEngineContext = getScheduler().getExecutor().getExecutionEngine().getExecutionEngineContext();
+		CustomReportType reportType = CustomReportType.parse(customReportType);
+		if (reportType == null) {
+			throw new ControllerServiceException(400, "Invalid report type: " + customReportType);
+		}
+		switch (reportType) {
+			case JUNIT:
+				JUnitXmlReportBuilder.Report report = new JUnitXmlReportBuilder(executionEngineContext, List.of(executionId)).buildJUnitXmlReport();
+				Response.ResponseBuilder response = Response.ok(new ByteArrayInputStream(report.getContent().getBytes()));
+				response.header("Content-Disposition", "attachment; filename=\"" + report.getMetadata().getFileName() + "\"");
+				return response.build();
+			default:
+				throw new ControllerServiceException(400, "Invalid report type: " + customReportType);
+		}
+	}
+
+	@Operation(description = "Returns the custom report for several executions")
+	@GET
+	@Path("/{id}/report/multi/{customReportType}")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.TEXT_PLAIN)
 	@Secured(right = "execution-read")
-	public String getCustomReport(@PathParam("id") String executionId, @PathParam("customReportType") String customReportType) {
+	public Response getCustomMultiReport(@PathParam("customReportType") String customReportType, @QueryParam("id") List<String> executionIds) {

```

Comment:
Not sure how the URL looks like for a list of execution IDs, shouldn't we use a POST request instead passing the array of IDs as request payload?

Reply:
Maybe the POST is better here, but for now I still kep the GET with `ids` query parameter (but now it is the `String` with `;` separator).